### PR TITLE
Additional work on tracing

### PR DESCRIPTION
**Pull request by jherico**
* Fix missing stat tracker for agents
* Migrate tracing categories to a parallel set of QLoggingCategories
* Set up a default set of tracing categories
* When writing a tracing file, if a relative path is provided, put the trace output relative the user's documents directory
* Add SAMPLE_XXX tracing macros to do sampling of values, rather than recording them every time

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1af9cc6dca4ded5e500d62a964bcb24225dc4f15 available at: https://gist.github.com/a0021cafbf4142bfb2ea0b013454c66b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9225-1af9cc6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9225-1af9cc6.dmg

**Comment by AndrewMeadows** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1af9cc6dca4ded5e500d62a964bcb24225dc4f15 available at: https://gist.github.com/a0021cafbf4142bfb2ea0b013454c66b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9225-1af9cc6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9225-1af9cc6.dmg

**Comment by OmegaHeron** [comment]
Will smoke test once build succeeds.  Anything specific beyond smoke test to look for?

**Comment by AndrewMeadows** [comment]
This failed for bad conflict resolution in Application.cpp.

**Comment by hifi-gustavo** [comment]
Build console output for commit: b7e6a8ea80704b61d7a2486bb54edf648f52dc2e available at: https://gist.github.com/a0021cafbf4142bfb2ea0b013454c66b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9225-b7e6a8e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9225-b7e6a8e.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 88e43a09572d0f08818edc4e5ff2b2ce14a6e1fc available at: https://gist.github.com/a0021cafbf4142bfb2ea0b013454c66b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9225-88e43a0.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9225-88e43a0.dmg

**Comment by samcake** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 88e43a09572d0f08818edc4e5ff2b2ce14a6e1fc available at: https://gist.github.com/a0021cafbf4142bfb2ea0b013454c66b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9225-88e43a0.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9225-88e43a0.dmg

File: domain-server/src/DomainServer.cpp [hunk 1]
```diff
@@ -75,6 +76,7 @@ DomainServer::DomainServer(int argc, char* argv[]) :
     parseCommandLine();
 
     DependencyManager::set<tracing::Tracer>();
+    DependencyManager::set<StatTracker>();
```
Comment (birarda) [inline domain-server/src/DomainServer.cpp]: Have we seen anything in the DomainServer that suggests that this would be required?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2f8af30a8a08b788b6cd27f1559c036d28c12c6d available at: https://gist.github.com/a0021cafbf4142bfb2ea0b013454c66b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9225-2f8af30.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9225-2f8af30.dmg

**Comment by OmegaHeron** [comment]
Windows 10 - i5 6500 CPU - GTX970 GPU

General smoke test and CPU check = pass.

File: libraries/shared/src/Profile.cpp [hunk 2]
```diff
@@ -8,6 +8,21 @@
...
+Q_LOGGING_CATEGORY(trace_resource, "trace.resource")
+Q_LOGGING_CATEGORY(trace_resource_network, "trace.resource.network")
+Q_LOGGING_CATEGORY(trace_resource_parse, "trace.resource.parse")
+Q_LOGGING_CATEGORY(trace_simulation, "trace.simulation")
+Q_LOGGING_CATEGORY(trace_simulation_animation, "trace.simulation.animation")
+Q_LOGGING_CATEGORY(trace_simulation_physics, "trace.simulation.physics")
+
+void foo() {
+    randFloat();
+}
```
Comment (birarda) [inline libraries/shared/src/Profile.cpp]: ?

Comment (jherico) [inline libraries/shared/src/Profile.cpp]: !

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2d735883613741d78b928b9e143f72c7dfffab23 available at: https://gist.github.com/a0021cafbf4142bfb2ea0b013454c66b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9225-2d73588.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9225-2d73588.dmg

**Pull request merged**